src/TestInfrastructure/Helpers/MefTestHelpers.cs

```diff
@@ -177,13 +176,46 @@ public static void Compose(Assembly exportingAssembly, params object[] importers
 
             var catalog = new AssemblyCatalog(exportingAssembly);
 
-            using var container = new CompositionContainer(catalog);
-            container.Compose(batch);
+            DoCompose(catalog, batch);
+        }
+
+        private static void DoCompose(ComposablePartCatalog catalog, CompositionBatch batch)
+        {
+            var container = new CompositionContainer(catalog);
+
+            try
+            {
+                container.Compose(batch);
+
+            }
+            finally
+            {
+                SafeDispose(container);
+            }
+        }
```

Comment:
Long-hand version of a `using` statement, with a clean-up step that ignores errors during disposal.
If the composition failed then the original `CompositionException` will be re-raised when the `finally` block has completed.